web/sites/default/settings.php

```diff
@@ -53,21 +53,34 @@
   $config['search_api.server.solr']['name'] = 'Lagoon Solr - Environment: ' . getenv('LAGOON_PROJECT');
 }
 
-### Lagoon Redis connection.
-if (getenv('LAGOON')){
-  $settings['redis.connection']['interface'] = 'PhpRedis';
-  $settings['redis.connection']['host'] = getenv('REDIS_HOST') ?: 'redis';
-  $settings['redis.connection']['port'] = 6379;
+// Redis configuration.
+if (getenv('LAGOON') && (getenv('ENABLE_REDIS'))) {
+  $redis = new \Redis();
+  $redis_host = getenv('REDIS_HOST') ?: 'redis';
+  $redis_port = getenv('REDIS_SERVICE_PORT') ?: 6379;
+  try {
+    if (drupal_installation_attempted()) {
+      # Do not set the cache during installations of Drupal
+      throw new \Exception('Drupal installation underway.');
+    }
+
+    $redis->connect($redis_host, $redis_port);
+    $response = $redis->ping();
+
+    if (strpos($response, 'PONG') === FALSE) {
+      throw new \Exception('Redis could be reached but is not responding correctly.');
+    }
+
+    $settings['redis.connection']['interface'] = 'PhpRedis';
+    $settings['redis.connection']['host'] = $redis_host;
+    $settings['redis.connection']['port'] = $redis_port;
+    $settings['cache_prefix']['default'] = getenv('REDIS_CACHE_PREFIX') ?: getenv('LAGOON_PROJECT') . '_' . getenv('LAGOON_GIT_SAFE_BRANCH');
 
-  $settings['cache_prefix']['default'] = getenv('LAGOON_PROJECT') . '_' . getenv('LAGOON_GIT_SAFE_BRANCH');
-
-  # Do not set the cache during installations of Drupal.
-  if (!drupal_installation_attempted() && extension_loaded('redis')) {
     $settings['cache']['default'] = 'cache.backend.redis';
 
     // Include the default example.services.yml from the module, which will
     // replace all supported backend services (that currently includes the cache tags
-    // checksum service and the lock backends, check the file for the current list).
+    // checksum service and the lock backends, check the file for the current list)
```

Comment:
unrelated change? Drupal coding standards mandate the use of fullstops.

Reply:
Fixed.